### ISSUE TITLE
ttl.math.reduce_{sum,max}(inp, <numeric scalar>, dims=[0|1]) returns wrong values on multi-tile inputs

### ISSUE THREAD

**Issue by brnorris03** [tenstorrent/tt-lang#594]
The numeric-scalar reduce API added in #589 produces incorrect output when the reduced dim leaves more than one output tile (i.e. the *non*-reduced dim spans multiple tiles). Single-tile inputs and the older tile-form scaler (`reduce(inp, scaler_tile, ...)`) are correct in these configurations.

**Observed:** `actual ≈ expected × 1.6` per output element. On `scaler=0.25`, `inp_shape=(2,2) tiles`, `dims=[0]`, bf16, `max_diff=2.97`.

**Standalone repro** (~5s):

```python
import torch, ttl, ttnn

TILE = 32

@ttl.operation(grid=(1, 1))
def reduce_kernel(inp, out):
    inp_dfb = ttl.make_dataflow_buffer_like(inp, shape=(2, 2), block_count=2)
    out_dfb = ttl.make_dataflow_buffer_like(out, shape=(1, 2), block_count=2)

    @ttl.compute()
    def compute_fn():
        with inp_dfb.wait() as inp_blk, out_dfb.reserve() as out_blk:
            out_blk.store(ttl.math.reduce_sum(inp_blk, 0.25, dims=[0]))

    @ttl.datamovement()
    def dm_read():
        with inp_dfb.reserve() as blk:
            ttl.copy(inp[0:2, 0:2], blk).wait()

    @ttl.datamovement()
    def dm_write():
        with out_dfb.wait() as blk:
            ttl.copy(blk, out[0:1, 0:2]).wait()

device = ttnn.open_device(device_id=0)
torch.manual_seed(0xBEEF)
inp_t = (torch.rand(2*TILE, 2*TILE, dtype=torch.bfloat16) - 0.5) * 4.0
out_t = torch.zeros(TILE, 2*TILE, dtype=torch.bfloat16)
inp = ttnn.from_torch(inp_t, layout=ttnn.TILE_LAYOUT, device=device, memory_config=ttnn.L1_MEMORY_CONFIG)
out = ttnn.from_torch(out_t, layout=ttnn.TILE_LAYOUT, device=device, memory_config=ttnn.L1_MEMORY_CONFIG)
reduce_kernel(inp, out)

expected = inp_t.float().sum(dim=0, keepdim=True) * 0.25
actual   = ttnn.to_torch(out)[:1, :].float()
print(f"max_diff = {(actual - expected).abs().max().item():.4f}")
ttnn.close_device(device)
```

**Scope** (bisected against pre-#589 commit `e4248ab7`; the numeric-scalar API itself doesn't exist there, so only the post-#589 behavior is directly testable):
- `dims=[0]` / `dims=[1]` multi-tile scalar-form → **broken, introduced by #589**
- `dims=[0]` / `dims=[1]` multi-tile *tile-form* → correct on both pre- and post-#589 (rules out the underlying reduce kernel; the bug is in the scalar-form post-reduce `mul_unary` path)
- `dims=[0,1]` multi-tile any form → pre-existing bug (separate issue to file)
- Single-tile non-1.0 scaler `dims=[0,1]` → pre-existing bug (separate issue to file)

**Likely root cause:** the post-reduce `mul_unary` driven by `ttl.reduce_scalar_multiplier` is applied to only a subset of result tiles when the output spans multiple tiles. Single-output-tile cases (`dims=[0,1]` collapses to one tile) hide this on the scalar form.

**Coverage:** `test/python/test_reduce_scalar.py::test_scalar_reduce_multi_tile` cross-product hits this; 24 cases currently fail. Suggest `xfail` markers referencing this issue until the fix lands.
